astropy/coordinates/solar_system.py

```diff
@@ -246,12 +256,37 @@ def get_body_barycentric(body, time, ephemeris=None):
             # passed in a kernel chain
             kernel_spec = body
 
-        cartesian_position_body = sum([kernel[pair].compute(jd1, jd2)
-                                       for pair in kernel_spec])
-
-        barycen_to_body_vector = u.Quantity(cartesian_position_body, unit=u.km)
-
-    return CartesianRepresentation(barycen_to_body_vector)
+        body_posvel_bary = np.zeros((2 if get_velocity else 1, 3) +
+                                    getattr(jd1, 'shape', ()))
+        for pair in kernel_spec:
+            spk = kernel[pair]
+            if spk.data_type == 3:
+                # Type 3 kernels contain both position and velocity.
+                posvel = kernel.compute(jd1, jd2)
```

Comment:
The `kernel` we have here is comes from `_get_kernel` and is, in fact, something returned by `SPK.open()`. I could rename, but then we are breaking backwards compatibility with the public `solar_system.get_kernel` method.